hack/e2e-pool-test.sh

```diff
@@ -180,6 +224,12 @@ FAKE_POOL_NAME=fake-pool
 oc get clusterpool ${REAL_POOL_NAME} -o json \
   | jq '.spec.annotations["hive.openshift.io/fake-cluster"] = "true" | .metadata.name = "'${FAKE_POOL_NAME}'" | .spec.size = 4' \
   | oc apply -f -
+
+# Add customization to REAL POOL
+NEW_CLUSTER_NAME="cdc-$(cat /dev/urandom | tr -dc 'a-z' | fold -w 8 | head -n 1)"
+create_customization "cdc-test" "${CLUSTER_NAMESPACE}" "${NEW_CLUSTER_NAME}"
+oc patch cp -n $CLUSTER_NAMESPACE $REAL_POOL_NAME --type=merge -p '{"spec": {"inventory": [{"name": "cdc-test"}]}}'
```

Comment:
> IIUC this is going to cause the real pool to trash its existing cluster (which is probably still provisioning at this stage) and replace it with one based on your CDC, right?

Yes, because of pool version change.

> First, I remember us talking about doing the CDC thing on the fake pool instead. Curious why you decided against that, or if you tried it.

I wasn't familiar with fake clusterpool logic and I wanted to make sure it actually works. I still feel a bit uncomfortable with fake clusterpool from integration/system test point of view. I personally would have use it as tier 0 in the integration testing to avoid using unecessary resources, but did the additional real test.

> ...but if we're going to keep it on the real pool, we should either let the original (non-CDC) cluster finish installing, validate it, and then edit the pool to use the inventory; or we should just create the pool with the inventory initially.
I like the latter -- it's broader coverage of the code -- but it would add ~40m to the test... unless we do all of that to the fake cluster instead.

Following my comment above, I would suggest to refactor a bit the test flow:
- Create two clusterpools of size 0
- Create the fake clusterpools,  and resize to >0, make sure the fake tests pass first
- Resize the two clusterpool to size >0
Why two clusterpools? Because you are paying for the cluster that is running Hive and for the the two clusters that anyway will be created.

Well after this PR merged, we can start pushing DPTP team to use internal resources and then you wouldn't care about the execution time. Hypershift could have helped too with the cluster cost for running hive e2e tests. But that is a different discussion.